consul/datadog_checks/consul/consul.py

```diff
@@ -27,14 +28,58 @@
     distance,
 )
 
+from .metrics import METRIC_MAP
+
+try:
+    import datadog_agent
+except ImportError:
+    from datadog_checks.base.stubs import datadog_agent
+
+
 NodeStatus = namedtuple('NodeStatus', ['node_id', 'service_name', 'service_tags_set', 'status'])
 
 
-class ConsulCheck(AgentCheck):
+class ConsulCheck(OpenMetricsBaseCheck):
     def __init__(self, name, init_config, instances):
-        super(ConsulCheck, self).__init__(name, init_config, instances)
+        instance = instances[0]
+        self.url = instance.get('url')
+        if self.url is None:
+            raise ConfigurationError("`url` parameter is required.")
+
+        # Set the prometheus endpoint configuration
+        self.use_prometheus_endpoint = is_affirmative(instance.get('use_prometheus_endpoint', False))
+        instance.setdefault('prometheus_url', '')
+        if self.use_prometheus_endpoint:
+            # Check that the prometheus and the dogstatsd method are not both configured
+            if self._is_dogstatsd_configured():
+                raise ConfigurationError(
+                    'The DogStatsD method and the Prometheus method are both enabled. Please choose only one.'
+                )
```

Comment:
Are the supported dogstatsd metrics all available from the prometheus endpoint?

Reply:
Yes they should be all available https://www.consul.io/docs/agent/telemetry, even if we don't retrieve them all due to the tag issue